Simulastsharedtask (#1693)
pytorch/fairseq: examples/simultaneous_translation/docs/baseline.md

# **Baseline Simultaneous Translation**
---

> This is an instruction of training and evaluating a *wait-k* simultanoes LSTM model on MUST-C English-Gernam Dataset.

Reviewer: ```suggestion
This is an instruction of training and evaluating a *wait-k* simultaneous LSTM model on the MUST-C English-German dataset.
```

> This is an instruction of training and evaluating a *wait-k* simultaneous LSTM model on the MUST-C English-German dataset.

Reviewer: ```suggestion
You can find here instructions for training and evaluating a *wait-k* simultaneous LSTM model on the MUST-C English-German dataset.
```

First, segment wav files.
```shell 
python $FAIRSEQ/examples/simultaneous_translation/data/segment_wav.py \
    --datapath $DATA_ROOT
```
Similar to text-to-text model, train a Sentencepiecemodel, but only train on German
```Shell
python $FAIRSEQ/examples/simultaneous_translation/data/train_spm.py \
    --data-path $DATA_ROOT/data \
    --vocab-size 10000 \
    --max-frame 3000 \
    --model-type unigram \
    --lang $lang \
    --out-path .
```
## Training
```shell
mkdir -p checkpoints
CUDA_VISIBLE_DEVICES=1 python $FAIRSEQ/train.py data-bin/mustc_en_de \
    --save-dir checkpoints \
    --arch berard_simul_text_iwslt \
    --waitk-lagging 2 \
    --waitk-stride 10 \
    --input-feat-per-channel 40 \
    --encoder-hidden-size 512 \
    --output-layer-dim 128 \
    --decoder-num-layers 3 \
    --task speech_translation \
    --user-dir $FAIRSEQ/examples/simultaneous_translation
    --optimizer adam \
    --max-epoch 100 \
    --lr 0.001 \
    --clip-norm 5.0  \
    --max-sentences 128  \
    --log-format json \
    --log-interval 10 \
    --criterion cross_entropy_acc \
    --user-dir $FAIRSEQ/examples/simultaneous_translation
```

## Evaluation
---
### Evaluation Server

> For text translation models, the server is set up as follow give input file and reference file. 

Reviewer: ```suggestion
For text translation models, the server needs an input file and reference file. 
```

``` shell
python ./eval/server.py \
    --hostname localhost \
    --port 12321 \
    --src-file $DATA_ROOT/data/dev/txt/dev.en \
    --ref-file $DATA_ROOT/data/dev/txt/dev.de
```
For speech translation models, the input is the data direcrory.
``` shell
python ./eval/server.py \
    --hostname localhost \
    --port 12321 \
    --ref-file $DATA_ROOT \
    --data-type speech
```

### Decode and Evaluate with Client

> Once the server is set up, run client to evaluate translation quality and latency.

Reviewer: ```suggestion
Once the server is set up, run the client to evaluate translation quality and latency.
```
